raster-foundry/raster-foundry
# Add project analyses list view

## Overview

This PR:
 - adds project analysis list UI
 - updates analysis list endpoint to return analysis info with related project layer info and template info if `projectId` QP is passed in
 - updates `ToolRunDao` and added property tests for new sql

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

### Demo

<img width="1386" alt="screen shot 2019-02-07 at 5 49 54 pm" src="https://user-images.githubusercontent.com/16109558/52448237-41c67e80-2b01-11e9-9cae-14c1d2fd05d5.png">

<img width="1386" alt="screen shot 2019-02-07 at 5 50 24 pm" src="https://user-images.githubusercontent.com/16109558/52448252-4db24080-2b01-11e9-918d-5369de9850ff.png">



### Notes

For features in the actions list -- only the delete action is fully implemented. The settings actions redirects to a link, which does not have page content yet.


## Testing Instructions

 * Go to lab
 * Create an analysis named with `<projectLayerName - Date>` (or not, this is just to mock an analysis name when we create it from within a project) with one of your projects with scenes, and make sure the analysis result shows up in the lab map
 * Go into db. Update the analysis you just created: `template_id`, `project_id`, and `project_layer_id` (we need to manually update them to test this PR since populating these three columns when creating an analysis is currently not supported in the lab UI)
 * Go to `v2/project/<projectId>/<analyses>` UI, make sure you see the analysis you just created listed on the sidebar and tiles show up on the map
 * Make some more such kind of analyses and go to the project to check them out. Use the analysis visibility filter on top. Use the visibility switch on each sidebar item.
 * Use the delete analysis option in the actions and make sure it works.

Closes https://github.com/raster-foundry/raster-foundry/issues/4542
Closes https://github.com/raster-foundry/raster-foundry/issues/4593
